src/django/api/views/v1/opensearch_query_builder/opensearch_query_builder.py

```diff
@@ -0,0 +1,147 @@
+import copy
+from api.views.v1.opensearch_query_builder. \
+    opensearch_query_builder_interface import OpenSearchQueryBuilderInterface
+
+
+class OpenSearchQueryBuilder(OpenSearchQueryBuilderInterface):
+    def __init__(self):
+        self.default_query_body = {
+            'track_total_hits': 'true',
+            'size': 10,
+            'query': {'bool': {'must': []}},
+            'sort': []
+        }
+        self.query_body = copy.deepcopy(self.default_query_body)
+        self.default_fuzziness = 2
+        self.default_sort = 'name'
+        self.default_sort_order = 'asc'
+
+    def reset(self):
+        self.query_body = copy.deepcopy(self.default_query_body)
+
+    def add_size(self, size):
+        self.query_body['size'] = size
+
+    def add_match(self, field, value, fuzziness=None):
+        if fuzziness is None:
+            fuzziness = self.default_fuzziness
+        match_query = \
+            {'match': {field: {'query': value, 'fuzziness': fuzziness}}}
+        self.query_body['query']['bool']['must'].append(match_query)
+
+    def add_multi_match(self, query):
+        self.query_body['query']['bool']['must'].append({
+            'multi_match': {
+                'query': query,
+                'fields': ['name^2', 'address', 'description', 'name_local'],
+                'fuzziness': self.default_fuzziness
+            }
+        })
+
+    def add_terms(self, field, values):
+        if not values:
+            return self.query_body
+        terms_field = f'{field}.alpha_2' if field == 'country' \
+            else f'{field}.keyword'
+
+        existing_terms = next(
+            (item['terms'] for item in self.query_body['query']['bool']['must']
+             if 'terms' in item and terms_field in item['terms']), None)
+
+        if existing_terms:
+            existing_terms[terms_field].extend(values)
+        else:
+            self.query_body['query']['bool']['must'].append({
+                'terms': {terms_field: values}
+            })
+
+    def add_range(self, field, query_params):
+        min_value = query_params.get(f'{field}[min]')
+        max_value = query_params.get(f'{field}[max]')
+        min_value = int(min_value) if min_value else None
+        max_value = int(max_value) if max_value else None
+
+        range_query = {}
+        if min_value is not None:
+            range_query['gte'] = min_value
+        if max_value is not None:
+            range_query['lte'] = max_value
+
+        if range_query:
+            if field == 'number_of_workers':
```

Comment:
There are multiple parts of the code where we look for a specific field name to decide how to build a search query. Is there any way we can do something similar to what we do in `OpenSearchQueryDirector`, by abstracting the code from using specific field names and defining some kind of mapping to refer to?

Reply:
This is a good point, but I decided to not use field-name-method mapping here because of decreasing of the code readability. Some builder methods in `OpenSearchQueryBuilder` accept `filed` name so it is ok **for me** that these particular methods will act depending on `field` value they receive. In the opposite, I need to create and maintain separate `_add_number_of_workers_range` which I'd like to avoid (at least for now). I agree that this is leaky abstraction, and `OpenSearchQueryBuilder` shouldn't suppose to know about field names, but I'd like to stick with strict OpenSearch queries naming (eg `add_range` , `add_terms`, `add_geo_distance` etc.) to preserve single responsibility principle here and expand those methods if necessary.

Reply:
Sounds fair.